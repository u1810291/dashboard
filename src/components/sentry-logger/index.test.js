import React from 'react'
import renderer from 'react-test-renderer'
import SentryLogger from '.'

jest.mock('@sentry/browser')

// eslint-disable-next-line react/require-render-return
class TestErrorComponent extends React.Component {
  render() {
    // eslint-disable-next-line no-throw-literal
    throw 'test error'
    // eslint-disable-next-line no-unreachable
    return null
  }
}

describe('When no JS errors are caught in a child component', () => {
  it('renders children', () => {
    const component = renderer.create(
      <SentryLogger>
        <span>Component with no errors</span>
      </SentryLogger>
    )
    expect(component.toJSON()).toMatchSnapshot()
  })
})

describe('When a JS error is caught in a child component', () => {
  let spy = jest.spyOn(SentryLogger.prototype, 'componentDidCatch')
  let component

  beforeAll(() => {
    global.console.error = jest.fn(() => {})
    component = renderer
      .create(
        <SentryLogger>
          <TestErrorComponent />
        </SentryLogger>
      )
      .toJSON()
  })

  afterAll(() => {
    global.console.error.removeMock()
  })

  test('reports errors for child componnents', () => {
    expect(spy).toHaveBeenCalled()
  })

  test('renders the report feedback link', () => {
    expect(component).toMatchSnapshot()
  })
})
