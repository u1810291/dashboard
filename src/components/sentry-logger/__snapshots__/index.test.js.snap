// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`When a JS error is caught in a child component renders the report feedback link 1`] = `
<button
  onClick={[Function]}
>
  Report feedback
</button>
`;

exports[`When no JS errors are caught in a child component renders children 1`] = `
<span>
  Component with no errors
</span>
`;
